core/lib/zksync_core/src/api_server/tx_sender/mod.rs

```diff
@@ -850,12 +850,8 @@ impl TxSender {
             .estimate_gas_binary_search_iterations
             .observe(number_of_iterations);
 
-        let tx_body_gas_limit = cmp::min(
-            MAX_L2_TX_GAS_LIMIT,
-            ((upper_bound as f64) * estimated_fee_scale_factor) as u64,
-        );
-
-        let suggested_gas_limit = tx_body_gas_limit + additional_gas_for_pubdata;
+        let suggested_gas_limit =
+            ((upper_bound + additional_gas_for_pubdata) as f64 * estimated_fee_scale_factor) as u64;
```

Comment:
should we check for overflow here?

Reply:
The overflow is never expected here, assuming adequate value of `estimated_fee_scale_factor`. There are also checks at the bottom of the function that the returned gas limit does not go beyond batch gas limit